platform/macos/os_macos.h

```diff
@@ -79,6 +79,21 @@ class OS_MacOS : public OS_Unix {
 	virtual void delete_main_loop() override;
 
 public:
+	static inline const char *headless_args[] = {
+		"--headless",
+		"-h",
+		"--help",
+		"/?",
+		"--version",
+		"--dump-gdextension-interface",
+		"--dump-extension-api",
+		"--dump-extension-api-with-docs",
+		"--validate-extension-api",
+		"--convert-3to4",
+		"--validate-conversion-3to4",
+		"--doctool",
+	};
```

Comment:
It's fine for now but having to maintain this list here is sure going to lead to inconsistencies eventually when relevant args are added/removed in `main`.

Most of those aren't actually headless though unless used together with `--headless`. E.g. `--dump-gdextension-interface` still shows a window by default.

We have `Main::is_cmdline_tool()` for most of these which could maybe be used here, especially if we were to change logic to imply that `cmdline_tool` means headless. I believe we wanted to do that in the past, but there might have been hurdles with headless mode that stopped it. CC @Calinou who might remember.

Reply:
> We have Main::is_cmdline_tool() for most of these which could maybe be used here

This will not work until `Main::setup` is executed, and it will be called long after the check (when list is used in the app entry point to select which version of `OS` to use).

> E.g. --dump-gdextension-interface still shows a window by default.

Not on macOS, everything on this list can't show windows. It was added specifically to prevent ~windows~ app icon in the dock from popping up when command line tools are used.

Reply:
The list is not new, just moved from `godot_main_macos.mm`, to avoid duplication.

Reply:
> > E.g. --dump-gdextension-interface still shows a window by default.
> 
> Not on macOS, everything on this list can't show windows. It was added specifically to prevent windows from popping up when command line tools are used.

Any reason for this behavior to be macOS specific?

Note: I see we imply headless for `--doctool` already, but not the other ones.

Reply:
Some other options missing from the list most likely:
```
--import  # unless we're not sure it's 100% safe headless
--export*  # likewise
--install-android-build-template
--dump-gdextension-interface-json
--test
```

Reply:
Not sure what's the best way to handle this, it probably should be in `Main`, but there's an issue with the way command line arguments caused by macOS specific quirks:

- `OS_MacOS` need to know if it's headless on init to pick correct main loop for the app.
- File associations for GUI apps are handled by event, not command line arguments, so `Main::start` argument parsing logic can run before GUI app is finished starting, or it will miss the opened files.

I guess making a separate setup function in the `Main` (which will only set headless flag) might be better.

Reply:
In the future, we'd like to make `--import` and `--export` imply `--headless`. This was originally attempted years ago, but we reverted it due to various bugs that occur with headless export. This is one of the reasons I created https://github.com/godotengine/godot/pull/109146.